lib/v2/screens/claim_invite_code/claim_invite_code_screen.dart

```diff
@@ -0,0 +1,229 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:seeds/constants/app_colors.dart';
+import 'package:seeds/providers/notifiers/auth_notifier.dart';
+import 'package:seeds/providers/notifiers/settings_notifier.dart';
+import 'package:seeds/providers/services/http_service.dart';
+import 'package:seeds/v2/components/flat_button_long.dart';
+import 'package:seeds/v2/components/scanner/seeds_qr_code_scanner_widget.dart';
+import 'package:seeds/v2/components/text_form_field_light_custom.dart';
+import 'package:seeds/utils/invites.dart';
+import 'package:seeds/models/models.dart';
+import 'package:seeds/design/app_theme.dart';
+import 'interactor/claim_invite_code_bloc.dart';
+
+enum ClaimCodeStatus {
+  emptyInviteCode,
+  searchingInvite,
+  foundNoInvite,
+  foundClaimedInvite,
+  foundValidInvite,
+  networkError,
+}
+
+class ClaimInviteCodeScreen extends StatefulWidget {
+  final ValueSetter<String> resultCallBack;
+  final String inviteCode;
+  final Function onClaim;
+
+  const ClaimInviteCodeScreen({Key key, this.resultCallBack, this.inviteCode, this.onClaim})
+      : super(key: key);
+
+  @override
+  _ClaimInviteCodeScreenState createState() => _ClaimInviteCodeScreenState();
+}
+
+class _ClaimInviteCodeScreenState extends State<ClaimInviteCodeScreen> {
+  ClaimInviteCodeBloc _ClaimInviteCodeBloc;
+  final _keyController = TextEditingController();
+  final _formImportKey = GlobalKey<FormState>();
+  final GlobalKey _qrKey = GlobalKey(debugLabel: 'QR');
+  bool _handledQrCode = false;
+  ClaimCodeStatus status = ClaimCodeStatus.emptyInviteCode;
```

Comment:
all these variables must live in the page state

Reply:
Remove all of them expect for 
  ClaimInviteCodeBloc _ClaimInviteCodeBloc;
  final _keyController = TextEditingController();
  final GlobalKey _qrKey = GlobalKey(debugLabel: 'QR');
need it for UI.